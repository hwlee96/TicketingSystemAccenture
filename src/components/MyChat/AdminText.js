import React from "react";

class AdminText extends React.Component {
  render() {
    return (
      <div className="messageplusdateadmin">
        <div className="adminmessage">
          <p className="adminchattext">{this.props.text}</p>
        </div>
        <div className="timeofmessageadmin">
          <p className="timeofmessagetext">{this.props.time}</p>
        </div>
      </div>
    );
  }
}

export default AdminText;
