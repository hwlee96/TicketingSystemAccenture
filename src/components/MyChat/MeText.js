import React from "react";

class MeText extends React.Component {
  render() {
    return (
      <div className="messageplusdateclient">
        <div className="clientmessage firstclientmessage">
          <p className="clientchattext">{this.props.text}</p>
        </div>
        <div className="timeofmessageclient">
          <p className="timeofmessagetext">{this.props.time}</p>
        </div>
      </div>
    );
  }
}

export default MeText;
