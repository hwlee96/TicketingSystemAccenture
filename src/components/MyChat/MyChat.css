/* -----------------------------------------------*/
/*  GLOBAL TERMS:
    SMALL WORDS: color: #414141;
    BACKGROUND COLOR: background-color: #fcfcfc; #e5e5e5; #f0f0f0; #ecf0f1

*/
/* -----------------------------------------------*/

/* -----------------------------------------------*/
/* BASIC SETUP */
/* -----------------------------------------------*/
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: "Lato", "Arial", sans-serif;
}

html {
  background-color: #fff;
  color: #414141;
  font-family: "Lato", "Arial", sans-serif;
  font-weight: 300;
  font-size: 20px;
  text-rendering: optimizeLegibility;
}

/* -----------------------------------------------*/
/* CHAT BOX */
/* -----------------------------------------------*/
.chatbox {
  width: 380px;
}

.chatboxheading {
  background-color: #1160bc;
  padding: 1%;
  border-bottom: 1px solid #868686;
}

.chatboxtextheading {
  font-size: 100%;
  color: #ffffff;
  margin-left: 2%;
  margin: 1.5%;
  display: inline-block;
}

.chatboxbody {
  padding: 0 5%;
  border: 1px solid #868686;
  border-bottom: none;
  border-top: none;
  min-height: 380px;
  background-color: #ffffff;
}

.messageplusdateclient {
  float: right;
}
.timeofmessageclient {
  display: block;
  margin-bottom: 10%;
  margin-left: 47%;
}

.timeofmessagetext {
  font-size: 70%;
}

.clientmessage {
  background-color: rgba(141, 194, 255, 0.79);
  width: 250px;
  display: block;
  border-radius: 7px;
  color: #000000;
  padding: 3%;
  word-wrap: break-word;
}

.firstclientmessage {
  margin-top: 5%;
}

.clientchattext {
  font-size: 80%;
}

.messageplusdateadmin {
  float: left;
}

.timeofmessageadmin {
  display: block;
  margin-bottom: 10%;
  margin-left: 3%;
}

.adminmessage {
  display: block;
  width: 250px;
  background-color: #e8e8e8;
  padding: 3%;
  border-radius: 7px;
  color: #000000;
  word-wrap: break-word;
}

.adminchattext {
  font-size: 80%;
}

.clearchat {
  clear: both;
}

.reply-box {
  height: 100px;
  width: 100%;
  border: 1px solid #868686;
  display: inline-block;
  background-color: #ffffff;
}

.typemessageherebox {
  display: inline-block;
  float: left;
}

.textarea {
  border: none;
  overflow: auto;
  outline: none;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
  box-shadow: none;
  resize: none;
  font-size: 78%;
  color: #868686;
  padding: 2%;
}

.sendbutton {
  border: none;
  background: rgba(0, 0, 0, 0);
  cursor: pointer;
  text-transform: uppercase;
  font-weight: 400;
  font-size: 90%;
  color: #198d34;
  display: inline-block;
  float: right;
  margin-top: 5%;
  margin-right: 3.5%;
  margin-bottom: 0;
}

.internal-scrolling-chat {
  height: 450px;
  overflow-y: scroll;
}
