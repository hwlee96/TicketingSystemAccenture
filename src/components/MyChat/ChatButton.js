import React, { Component } from "react";
import MyChat from "./MyChat";
import Fab from "@material-ui/core/Fab";

import chatIcon from "./chat.svg";

const style = {
  margin: 0,
  top: "auto",
  right: 20,
  bottom: 20,
  left: "auto",
  position: "fixed",
  backgroundColor: "#416FBD",
  color: "#FFFFFF"
};

class ChatButton extends Component {
  constructor(props) {
    super(props);
    this.state = { show: false };

    this.toggleDiv = this.toggleDiv.bind(this);
  }

  toggleDiv = () => {
    const { show } = this.state;
    this.setState({ show: !show });
  };
  render() {
    return (
      <div>
        <div>
          {this.state.show && (
            <MyChat
              author={this.props.author}
              requestID={this.props.requestID}
            />
          )}
        </div>
        <Fab style={style} onClick={this.toggleDiv}>
          <img src={chatIcon} alt="chatIcon" />
          <i className="fas fa-comment" />
        </Fab>
      </div>
    );
  }
}
export default ChatButton;
