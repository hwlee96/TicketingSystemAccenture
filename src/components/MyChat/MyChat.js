import React from "react";
import firebase from "firebase/app";
import "./MyChat.css";
import MeText from "./MeText";
import AdminText from "./AdminText";

const style = {
  margin: 0,
  top: "auto",
  right: 20,
  bottom: 80,
  left: "auto",
  position: "fixed"
};

class MyChat extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      message: "",
      messages: []
    };
    this.updateMessage = this.updateMessage.bind(this);
    this.submitMessage = this.submitMessage.bind(this);
  }

  componentDidMount() {
    firebase
      .database()
      .ref(`messages/${this.props.requestID}/`)
      .on("value", snapshot => {
        const currentMessages = snapshot.val();
        if (currentMessages != null) {
          this.setState({
            messages: currentMessages
          });
        }
      });
  }

  updateMessage(e) {
    this.setState({ message: e.target.value });
  }
  submitMessage(e) {
    e.preventDefault();

    var today = new Date();
    var date =
      today.getFullYear() +
      "-" +
      (today.getMonth() + 1) +
      "-" +
      today.getDate();
    var time =
      today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
    var dateTime = date + " " + time;

    this.setState({ message: "" }); //clear the input
    const nextMessage = {
      id: this.state.messages.length,
      text: this.state.message,
      author: this.props.author,
      time: dateTime
    };
    // let list = Object.assign([], this.state.messages);
    // list.push(nextMessage);
    //
    // this.setState({ messages: list });

    firebase
      .database()
      .ref(`messages/${this.props.requestID}/` + nextMessage.id)
      .set(nextMessage);
  }

  render() {
    const currentMessage = this.state.messages.map((message, i) => {
      if (this.props.author === "client") {
        if (message.author !== "admin") {
          // sent by me
          return (
            <MeText text={message.text} time={message.time} key={message.id} />
          );
        } else {
          return (
            <AdminText
              text={message.text}
              time={message.time}
              key={message.id}
            />
          );
        }
      } else {
        if (message.author !== "admin") {
          // sent by me
          return (
            <AdminText
              text={message.text}
              time={message.time}
              key={message.id}
            />
          );
        } else {
          return (
            <MeText text={message.text} time={message.time} key={message.id} />
          );
        }
      }
    });
    return (
      <div style={style}>
        {/* {this.props.author === "joseph" ? (
          <h3>Ben (Admin)</h3>
          ) : (
          <h3>Joseph</h3>
        )} */}

        {/* chat history starts here */}
        <div className="chatbox">
          <div className="chatboxheading">
            <h1 className="chatboxtextheading">Chat Support</h1>
          </div>

          <div className="chatboxbody internal-scrolling-chat">
            {currentMessage}

            <div className="clearchat" />
          </div>

          <form onSubmit={this.submitMessage}>
            <div className="reply-box">
              <button type="submit" className="sendbutton">
                Send
              </button>
              <div className="typemessageherebox">
                <textarea
                  rows="4"
                  cols="35"
                  className="textarea"
                  placeholder="Type your message here..."
                  onChange={this.updateMessage}
                  value={this.state.message}
                />
              </div>
            </div>
          </form>
        </div>
        {/* chat history ends here */}
      </div>
    );
  }
}

export default MyChat;
