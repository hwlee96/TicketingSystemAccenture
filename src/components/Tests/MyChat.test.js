import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import database from "../firebase/firebaseExport";

import MyChat from "../MyChat/MyChat";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<MyChat />).toJSON();
  expect(tree).toMatchSnapshot();
});
