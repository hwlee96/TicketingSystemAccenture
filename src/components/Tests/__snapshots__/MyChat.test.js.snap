// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "bottom": 80,
      "left": "auto",
      "margin": 0,
      "position": "fixed",
      "right": 20,
      "top": "auto",
    }
  }
>
  <div
    className="chatbox"
  >
    <div
      className="chatboxheading"
    >
      <h1
        className="chatboxtextheading"
      >
        Chat Support
      </h1>
    </div>
    <div
      className="chatboxbody internal-scrolling-chat"
    >
      <div
        className="clearchat"
      />
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="reply-box"
      >
        <button
          className="sendbutton"
          type="submit"
        >
          Send
        </button>
        <div
          className="typemessageherebox"
        >
          <textarea
            className="textarea"
            cols="35"
            onChange={[Function]}
            placeholder="Type your message here..."
            rows="4"
            value=""
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
